packages/hardhat-core/test/internal/hardhat-network/provider/modules/hardhat.ts

```diff
@@ -2143,27 +2172,36 @@ describe("Hardhat module", function () {
             this.provider,
             "hardhat_setStorageAt",
             [DEFAULT_ACCOUNTS_ADDRESSES[0], "xyz", numberToRpcQuantity(99)],
-            'Errors encountered in param 1: Invalid value "xyz" supplied to : QUANTITY'
+            // TODO: https://github.com/NomicFoundation/edr/issues/104
+            `${
+              isEdr ? "" : "Errors encountered in param 1: "
+            }Invalid value "xyz" supplied to : QUANTITY`
           );
         });
 
         it("should reject a storage key that is greater than 32 bytes", async function () {
           const MAX_WORD_VALUE = 2n ** 256n;
-          await assertInvalidInputError(
+          await assertProviderError(
             this.provider,
             "hardhat_setStorageAt",
             [
               DEFAULT_ACCOUNTS_ADDRESSES[0],
               numberToRpcQuantity(MAX_WORD_VALUE + 1n),
               "0xff",
             ],
-            "Storage key must not be greater than or equal to 2^256. Received 115792089237316195423570985008687907853269984665640564039457584007913129639937."
+            // TODO: https://github.com/NomicFoundation/edr/issues/104
+            `Storage key must not be greater than or equal to 2^256. Received ${
+              isEdr
+                ? "0x10000000000000000000000000000000000000000000000000000000000000001"
+                : "115792089237316195423570985008687907853269984665640564039457584007913129639937"
```

Comment:
Why are the values allowed to differ between EDR and Hardhat?

Reply:
This is something I wanted to discuss:

In JS/TS there is the `BigInt` type which is a smart integer type that can grow to any size. When this error is created by Hardhat, the incoming hexadecimal string is first parsed to a `BigInt` and then a check occurs to see whether the key is too large.

I wonder whether we should actually be printing the decimal notation of the key or the original hex string?

@fvictorio & @alcuadrado, I would appreciate your input.

If we should be returning a decimal number, we'd need to decode the hexadecimal value to a decimal string value, as there is no equivalent of the `BigInt` in our codebase.

Reply:
I think the user would expect the original hex string, as that's what we are telling them that is invalid.

Reply:
In that case, Hardhat would need to change its behaviour to return a hexadecimal string instead of the decimal one.

Reply:
I resolved it as follows: https://github.com/NomicFoundation/hardhat/pull/4721/commits/052ebe85d8603e6a830bdf8751fdd72ad6800f63